internal/encoder/encoder.go

```diff
@@ -213,10 +213,14 @@ func Encode(val interface{}, opts Options) ([]byte, error) {
     }
 
     /* make a copy of the result */
-    ret = make([]byte, len(buf))
-    copy(ret, buf)
-
-    freeBytes(buf)
+    if option.CanSizeResue(cap(buf)) {
```

Comment:
maybe we can use a new `LimitedBytesPool` type and not export `CanSizeReuse` func